File: src/jserver/src/test/velvetdawn/test_map_creation.py
import velvet_dawn
from velvet_dawn.config import Config
from velvet_dawn.dao import app
from test.base_test import BaseTest


class TestMapCreation(BaseTest):
    def test_map_creation(self):
        with app.app_context():
            config = Config().set_map_size(20, 20)
            config.seed = 79221
            config.datapacks = ['civil-war']

            velvet_dawn.datapacks.init(config)
            velvet_dawn.map.new(config)

File: src/jserver/src/test/velvetdawn/map/TestMapCreation.java
package test.velvetdawn.map;

import org.junit.Test;
import test.BaseTest;
import velvetdawn.core.VelvetDawn;
import velvetdawn.core.models.config.Config;

import java.util.List;

import static junit.framework.TestCase.assertFalse;
import static junit.framework.TestCase.assertTrue;
import static org.junit.Assert.assertEquals;

public class TestMapCreation extends BaseTest {

    @Test
    public void test_map_creation() throws Exception {
        var config = new Config();
        config.map.height = 20;
        config.map.width = 20;
        config.seed = 79221;
        config.datapacks = List.of("civil-war");

        var velvetDawn = new VelvetDawn(config);
        velvetDawn.players.join("acdc", "");

        velvetDawn.map.generate();
    }
}